go/vt/vtadmin/api.go

```diff
@@ -1051,6 +1053,115 @@ func (api *API) GetShardReplicationPositions(ctx context.Context, req *vtadminpb
 	}, nil
 }
 
+// GetSrvKeyspace is part of the vtadminpb.VTAdminServer interface.
+func (api *API) GetSrvKeyspace(ctx context.Context, req *vtadminpb.GetSrvKeyspaceRequest) (*vtctldatapb.GetSrvKeyspacesResponse, error) {
+	span, ctx := trace.NewSpan(ctx, "API.GetSrvKeyspace")
+	defer span.Finish()
+
+	c, err := api.getClusterForRequest(req.ClusterId)
+	if err != nil {
+		return nil, err
+	}
+
+	if !api.authz.IsAuthorized(ctx, c.ID, rbac.SrvKeyspaceResource, rbac.GetAction) {
+		return nil, nil
+	}
+
+	return c.Vtctld.GetSrvKeyspaces(ctx, &vtctldatapb.GetSrvKeyspacesRequest{
+		Keyspace: req.Keyspace,
+		Cells:    req.Cells,
+	})
+}
+
+// GetSrvKeyspaces is part of the vtadminpb.VTAdminServer interface.
+func (api *API) GetSrvKeyspaces(ctx context.Context, req *vtadminpb.GetSrvKeyspacesRequest) (*vtadminpb.GetSrvKeyspacesResponse, error) {
+	span, ctx := trace.NewSpan(ctx, "API.GetSrvKeyspaces")
+	defer span.Finish()
+
+	clusters, _ := api.getClustersForRequest(req.ClusterIds)
+
+	var (
+		sks map[string]*vtctldatapb.GetSrvKeyspacesResponse = make(map[string]*vtctldatapb.GetSrvKeyspacesResponse)
+		wg  sync.WaitGroup
+		er  concurrency.AllErrorRecorder
+		m   sync.Mutex
+	)
+
+	for _, c := range clusters {
+		if !api.authz.IsAuthorized(ctx, c.ID, rbac.SrvKeyspaceResource, rbac.GetAction) {
+			continue
+		}
+
+		wg.Add(1)
+
+		go func(c *cluster.Cluster) {
+			defer wg.Done()
+
+			span, ctx := trace.NewSpan(ctx, "Cluster.GetSrvKeyspaces")
```

Comment:
IMO everything from here to the end of the inner goroutine is better placed in a cluster-level method (similar to [`GetShardReplicationPositions`](https://github.com/vitessio/vitess/pull/12702/files#diff-3e212bf9fa780da22b2bbce441ebb9ed37a8c590bfa017c13ecb6e8b245fd5e3R1033) above

Reply:
(this one is the "main thing" i was referring to in my top-level comment, which i realize now was potentially _extremely_ not clear)

Reply:
ooh, good idea. Will push that up!